scripts/upgrade_chromedriver.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Upgrades the chromedriver dev-dependency to the one supported by the version of Google Chrome
+ * installed on the machine.
+ *
+ * Usage: node scripts/upgrade_chromedriver.js [--install]
+ */
+
+/* eslint no-restricted-syntax: 0 */
+const { execSync, spawnSync } = require('child_process');
+const { createReadStream, createWriteStream, unlinkSync, renameSync, existsSync } = require('fs');
+const { createInterface } = require('readline');
+
+if (!process.argv.includes(__filename)) {
+  console.error('Usage: node scripts/upgrade_chromedriver.js [--install]');
+  process.exit(1);
+}
+
+const versionCheckCommands = [];
+
+switch (process.platform) {
+  case 'win32':
+    versionCheckCommands.push(
+      'powershell "(Get-Item \\"$Env:Programfiles/Google/Chrome/Application/chrome.exe\\").VersionInfo.FileVersion"'
+    );
+    break;
+
+  case 'darwin':
+    versionCheckCommands.push(
+      '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome --version'
+    );
+    break;
+
+  default:
+    versionCheckCommands.push(
+      ...[
+        '/usr/bin',
+        '/usr/local/bin',
+        '/usr/sbin',
+        '/usr/local/sbin',
+        '/opt/bin',
+        '/usr/bin/X11',
+        '/usr/X11R6/bin',
+      ].flatMap((loc) =>
+        [
+          'google-chrome --version',
+          'google-chrome-stable --version',
+          'chromium --version',
+          'chromium-browser --version',
+        ].map((cmd) => `${loc}/${cmd}`)
+      )
+    );
+}
+
+let versionCheckOutput;
+versionCheckCommands.some((cmd) => {
+  try {
+    console.log(cmd);
+    versionCheckOutput = execSync(cmd, { encoding: 'utf8' })?.trim?.();
```

Comment:
The Optional chaining syntax breaks the CI tests

Reply:
hmm, good spot. I wonder if we just need to add this file to the paths that get transpiled - not sure any of our scripts get transpiled by default.

Reply:
merged node10 compatible changes